### ISSUE TITLE
File Nesting: Uppercase file names are not matched correctly

### ISSUE THREAD

**Issue by wopian** [microsoft/vscode#142390]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.64.0
- OS Version: macOS 12.2

Steps to Reproduce:

A) Nesting `Dockerfile` with a capitalised pattern (`Dockerfile`)

1. Add `"package.json": "Dockerfile,.dockerignore"` to `explorer.experimental.fileNesting.patterns`
2. `Dockerfile` is not nested into `package.json`

B) Nesting `Dockerfile` with a lowercase pattern (`dockerfile`)

1. Add `"package.json": "dockerfile,.dockerignore"`
2. `Dockerfile` is nested into `package.json` (at least on macOS with a case-insensitive OS)

C) Nesting files into `Dockerfile` with `Dockerfile` pattern

1. Add `"Dockerfile": ".dockerignore"`
2. `.dockerignore` is not nested

D) Nesting files into `Dockerfile` with `dockerfile` pattern

1. Add `"dockerfile": ".dockerignore"`
2. `.dockerignore` is nested into `Dockerfile`

<details><summary>Original issue reproduction:</summary>

A) Nesting into Dockerfile:

1. Add `"Dockerfile": ".dockerignore"` to `explorer.experimental.fileNesting.patterns`
2. `.dockerignore` is not nested into `Dockerfile

B) Nesting Dockerfile into another file:

1. Add `"package.json": "Dockerfile,.dockerignore"`
2. Only `.dockerignore` is nested into `package.json`

Note that this issue does not apply to all extension-less files - creating a random file e.g `env` and adding it to `"package.json": "env"` nests it correctly.
</details>

**Comment by JacksonKearl** [comment]
Interesting, thanks for the report.

**Comment by rafasoares** [comment]
@JacksonKearl thanks for looking into this! I just want to clarify if this issue is supposed to be fixed on 1.66.0 stable.

If it is, then the bug is still present (I'm on Version: 1.66.0 - Commit: e18005f0f1b33c29e81d732535d8c0e47).

I'm testing with these settings:
```json
  "explorer.experimental.fileNesting.patterns": {
      "Gemfile": "Gemfile.lock"
  }
```

If I change it to this, the nesting works (notice only the parent needs to be lowercase):
```json
  "explorer.experimental.fileNesting.patterns": {
      "gemfile": "Gemfile.lock"
  }
```

Tested on macOS 11.6.5.

Looking at the code, it seems `this.getPlatformAwareName()` also needs to be applied to `parentPattern`.

Let me know if I need to create a new issue.
